sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/RestrictionTracker.java

```diff
@@ -99,6 +100,10 @@
    */
   public abstract void checkDone() throws IllegalStateException;
 
+  public boolean isBounded() throws NotImplementedException {
+    throw new NotImplementedException("isBounded is not implemented.");
```

Comment:
We should add a comment stating the unbounded restriction trackers can only be used with `@UnboundedPerElement` SDFs. We should add a comment on each restriction tracker whether it could be unbounded.

Reply:
Not relevant anymore.